src/dmd/cparse.d

```diff
@@ -2488,28 +2488,31 @@ final class CParser(AST) : Parser!AST
     /*************************
      * __attribute__ parser
      * https://gcc.gnu.org/onlinedocs/gcc/Attribute-Syntax.html
-     * attribute-specifier:
-     *    __attribute__ (( attribute-list ))
+     * gnu-attributes:
+     *   gnu-attributes gnu-attribute-specifier
      *
-     * attribute-list:
-     *    attribute (opt)
-     *    attribute-list , attribute
+     * gnu-attribute-specifier:
+     *    __attribute__ (( gnu-attribute-list ))
      *
-     * attribute:
-     *    attribute-name
-     *    attribute-name ( identifier )
-     *    attribute-name ( identifier , expression-list )
-     *    attribute-name ( expression-list (opt) )
+     * gnu-attribute-list:
+     *    gnu-attribute (opt)
+     *    gnu-attribute-list , gnu-attribute
      *
-     * attribute-name:
+     * gnu-attribute:
+     *    gnu-attribute-name
+     *    gnu-attribute-name ( identifier )
+     *    gnu-attribute-name ( identifier , expression-list )
+     *    gnu-attribute-name ( expression-list (opt) )
+     *
+     * gnu-attribute-name:
      *    keyword
      *    identifier
      *
      * expression-list:
-     *    constant-expression
-     *    expression-list constant-expression
+     *    assignment-expression
```

Comment:
This needs to be a constant-expression, as it must be evaluated at compile time.

Reply:
Right, but this is not enforced at a grammar/parser level, but at semantic.

Though to be honest, `constant-expression` just forwards to `assignment-expression` anyway in the cparse implementation.
```
private AST.Expression cparseConstantExp()
{
    return cparseAssignExp();
}
```